backend/utils/dns/index.js

```diff
@@ -37,8 +45,131 @@ function isPublicDNSName(v) {
   )
 }
 
+/**
+ * Is the given hostname a valid DNS name
+ *
+ * @param v {string} - The string to check
+ * @returns {boolean} - if the given name is a valid DNS name
+ */
+function isValidDNSName(v) {
```

Comment:
nit: as opposed to duplicating this method's code on the backend and frontend, it could be shared under packages/utils

Reply:
I didn't know this existed.  It's a pretty trivial function so I think I'll skip it for this PR but I'll keep this in mind for the future.